File: next-js/app/java_franca/graphics/views/page.ts
import {View} from "./view";
import {Router} from "../device/router";
import {Painter} from "../device/painter";

export class Page {
  router: Router | null = null;
  views: View[] = [];
  nextPage: Page | null = null;

  constructor(router: Router) {
    this.router = router;
  }

  destroy(): void {
    // очистка массива views
    this.views = [];
  }

  paint(painter: Painter): void {
    for (let i = 0; i < this.views.length; i++) {
      this.views[i].paint(this.router!, painter, this);
    }
  }

  handlePointerDown(pointedX: number, pointedY: number, buttonNumber: number): void {
    for (let i = 0; i < this.views.length; i++) {
      this.views[i].handlePointerDown(
        this.router!,
        this,
        0, // painterX
        0, // painterY
        pointedX,
        pointedY,
        buttonNumber
      );
    }
  }
}

File: engine/franca/java/graphics/views/Page.java
package franca.java.graphics.views;

import contracted.franca.java.TranspilableClass;
import contracted.franca.java.ContractedArray;
import franca.java.graphics.device.Painter;
import franca.java.graphics.device.Router;

public class Page extends TranspilableClass {

  Router router = null;

  public ContractedArray<View> views = new ContractedArray<View>();

  public Page nextPage = null;

  public Page(Router router) {
    this.router = router;
  }

  @Override
  public void destroy() {
    delete(views);
    views = null;
    super.destroy();
  }

  public void paint(Painter painter) {
    for (int i = 0; i < views.size(); i++) {
      views.get(i).paint(router, painter, this);
    }
  }

  public void handlePointerDown(float pointedX, float pointedY, int buttonNumber) {
    for (int i = 0; i < views.size(); i++) {
      views.get(i).handlePointerDown(router, this, 0f, 0f, pointedX, pointedY, buttonNumber);
    }
  }
}
